### PR TITLE
share code

### PR DESCRIPTION
**Pull request by orgrinberg**
share code via pull request

### CODE REVIEW

File: project1/src/udpclient.c [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <netinet/in.h>
+
+#define PORT 4080 //The port number.
```
Comment (idanelias) [inline project1/src/udpclient.c]: defines need to be in header file

File: project1/src/udpclient.c [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+#define MAXLINE 1024 //The maximum lengh to the string.
```
Comment (idanelias) [inline project1/src/udpclient.c]: defines need to be in header file

File: project1/src/udpclient.c [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+int sendsrt(char *buf ,int sockfd,struct sockaddr_in servaddr)	//Send to the server.
+{
+	sendto(sockfd, (const char *)buf, strlen(buf), MSG_CONFIRM, (const struct sockaddr *) &servaddr, sizeof(servaddr));
```
Comment (idanelias) [inline project1/src/udpclient.c]: I can understand why MSG_CONFIRM will be at the server side, but why should this flag be set in client side.

File: project1/src/udpclient.c [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+// Driver code
+int main(char *argv[]) {
+	int sockfd;
+	char buf[MAXLINE];
+	struct sockaddr_in servaddr;
+
+	// Creating socket file descriptor
+	if ( (sockfd = socket(AF_INET, SOCK_DGRAM, 0)) == -1 )
```
Comment (idanelias) [inline project1/src/udpclient.c]: function creating and returning a file descriptor will be cleaner.

File: project1/src/udpclient.c [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+	{
+		perror("socket creation failed");
+		return 0;
+	}
+
+	memset(&servaddr, 0, sizeof(servaddr));
```
Comment (idanelias) [inline project1/src/udpclient.c]: function defining server address will be cleaner.

File: project1/src/udpclient.c [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+		}
+		else {
+			printf("server not found\n");
+		}
+		
+		receives(buf,sockfd,servaddr);//Receives from the server.
+	}
+	
+
+	close(sockfd);
```
Comment (idanelias) [inline project1/src/udpclient.c:78]: code unreachable, the only way to exit the while loop is through a "return".

File: project1/src/udpserver.c [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <netinet/in.h>
+
+#define PORT 4080 //The port number.
```
Comment (idanelias) [inline project1/src/udpserver.c]: defines should be in the header

File: project1/src/udpserver.c [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+#define MAXLINE 1024//The maximum lengh to the string.
+
```
Comment (idanelias) [inline project1/src/udpserver.c:3]: define should be in the header

File: project1/src/udpserver.c [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+// Driver code
+int main() {
+	int sockfd;
+	char buf[MAXLINE];
+	struct sockaddr_in servaddr, cliaddr;
+	int len, n;
+
+	// Creating socket file descriptor
+	if ( (sockfd = socket(AF_INET, SOCK_DGRAM, 0)) == -1 )
```
Comment (idanelias) [inline project1/src/udpserver.c]: function creating and returning a file descriptor will be cleaner.

File: project1/src/udpserver.c [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+	{
+		perror("socket creation failed");
+		return 0;
+	}
+
+	memset(&servaddr, 0, sizeof(servaddr));
```
Comment (idanelias) [inline project1/src/udpserver.c]: function defining server address will be cleaner.

File: project1/src/udpserver.c [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+	memset(&cliaddr, 0, sizeof(cliaddr));
+
+	// Filling server information
+	servaddr.sin_family = AF_INET; // IPv4
+	servaddr.sin_addr.s_addr = INADDR_ANY;
+	servaddr.sin_port = htons(PORT);
+
+	// Bind the socket with the server address
+	if ( bind(sockfd, (const struct sockaddr *)&servaddr, sizeof(servaddr)) == -1 )
```
Comment (idanelias) [inline project1/src/udpserver.c]: function binding the socket will be much cleaner

File: project1/include/in.h [hunk 12]
```diff
@@ -0,0 +1,637 @@
+/* Copyright (C) 1991-2021 Free Software Foundation, Inc.
```
Comment (idanelias) [inline project1/include/in.h]: standard libs doesn't need to be inside the include.

File: project1/include/inet.h [hunk 13]
```diff
@@ -0,0 +1,105 @@
+/* Copyright (C) 1997-2021 Free Software Foundation, Inc.
```
Comment (idanelias) [inline project1/include/inet.h]: standard libs doesn't need to be inside the include.

File: project1/include/socket.h [hunk 14]
```diff
@@ -0,0 +1,274 @@
+/* Declarations of socket constants, types, and functions.
```
Comment (idanelias) [inline project1/include/socket.h]: standard libs doesn't need to be inside the include.

File: project1/include/stdio.h [hunk 15]
```diff
@@ -0,0 +1,877 @@
+/* Define ISO C stdio on top of C++ iostreams.
```
Comment (idanelias) [inline project1/include/stdio.h]: standard libs doesn't need to be inside the include.

File: project1/include/stdlib.h [hunk 16]
```diff
@@ -0,0 +1,1028 @@
+/* Copyright (C) 1991-2021 Free Software Foundation, Inc.
```
Comment (idanelias) [inline project1/include/stdlib.h]: standard libs doesn't need to be inside the include.

File: project1/include/string.h [hunk 17]
```diff
@@ -0,0 +1,525 @@
+/* Copyright (C) 1991-2021 Free Software Foundation, Inc.
```
Comment (idanelias) [inline project1/include/string.h]: standard libs doesn't need to be inside the include.

File: project1/include/types.h [hunk 18]
```diff
@@ -0,0 +1,232 @@
+/* Copyright (C) 1991-2021 Free Software Foundation, Inc.
```
Comment (idanelias) [inline project1/include/types.h]: standard libs doesn't need to be inside the include.

File: project1/include/unistd.h [hunk 19]
```diff
@@ -0,0 +1,1191 @@
+/* Copyright (C) 1991-2021 Free Software Foundation, Inc.
```
Comment (idanelias) [inline project1/include/unistd.h]: standard libs doesn't need to be inside the include.

**Review by idanelias — CHANGES_REQUESTED** [review]
overall, I think you've done a great work! see the comments in the files and fix them, especially the functions. this will help you see that some functions a repeatable between modules and can be taken to there own "udp_handler" module. but that something for project2 you'll need it more when we'll get there.

**Comment by orgrinberg** [comment]
all comment were implemented

File: project1/src/udpclient.c [hunk 20]
```diff
@@ -0,0 +1,82 @@
...
+		printf("server not found\n");
+		return 0;
+	}
+
+	memsetfunc(servaddr);
+
+	// Filling server information
+	servaddr.sin_family = AF_INET;
+	servaddr.sin_port = htons(PORT);
+	servaddr.sin_addr.s_addr = INADDR_ANY;
```
Comment (idanelias) [inline project1/src/udpclient.c]: lines 59-61 can be joined to memsetfunc to make this function more useful. with the purpose of initiating a sockaddr_in struct.

File: project1/src/udpserver.c [hunk 21]
```diff
@@ -0,0 +1,88 @@
...
+{
+	if ( bind(sockfd, (const struct sockaddr *)&servaddr, sizeof(servaddr)) == -1 )
+	{
+		perror("bind failed");
+		return 0;
+	}
+	return 1;
+}
+
+void memsetfunc(struct sockaddr_in addr)
```
Comment (idanelias) [inline project1/src/udpserver.c:21]: expand this function.

**Review by idanelias — CHANGES_REQUESTED** [review]
great job! one little thing to make more sense and we are there!

**Review by idanelias — APPROVED** [review]
Great job!

**Pull request merged**